app/Http/Controllers/Backend/Transport/StationController.php

```diff
@@ -32,19 +32,17 @@ public function __construct(?StationRepository $stationRepository = null) {
      * @return Collection
      */
     public static function getLatestArrivals(User $user, int $maxCount = 5): Collection {
-        $groupAndSelect = [
-            'train_stations.id', 'train_stations.ibnr', 'train_stations.name',
-            'train_stations.latitude', 'train_stations.longitude', 'train_stations.rilIdentifier',
-        ];
-        return DB::table('train_checkins') //TODO: return Station objects
-                 ->join('train_stopovers', 'train_checkins.destination_stopover_id', '=', 'train_stopovers.id')
-                 ->join('train_stations', 'train_stopovers.train_station_id', '=', 'train_stations.id')
-                 ->where('train_checkins.user_id', $user->id)
-                 ->groupBy($groupAndSelect)
-                 ->select($groupAndSelect)
-                 ->orderByDesc(DB::raw('MAX(train_checkins.arrival)'))
-                 ->limit($maxCount)
-                 ->get();
+        return DB::table('train_checkins')
+                ->join('train_stopovers', 'train_checkins.destination_stopover_id', '=', 'train_stopovers.id')
+                ->select('train_stopovers.train_station_id as station_id')
+                ->where('train_checkins.user_id', $user->id)
+                ->groupBy('station_id')
+                ->orderByDesc(DB::raw('MAX(train_checkins.arrival)'))
+                ->limit($maxCount)
+                ->pluck('station_id')
+                ->map(function (int $id) {
+                    return Station::where('id', $id)->first();
```

Comment:
This would result in an additional `$maxCounts` SQL queries, which would not scale well on the production instance. :(